### PR TITLE
Add API sample code to AccessibilityGuideline

### LINKED ISSUE #102158: Add API sample code to AccessibilityGuideline
It should showcase how to use the individual guidelines in a test.

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/flutter/issues/102158

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_test/lib/src/accessibility.dart [hunk 1]
```diff
@@ -58,6 +58,34 @@ class Evaluation {
...
+///   final SemanticsHandle handle = tester.ensureSemantics();
+///   await tester.pumpWidget(const MaterialApp(home: HomePage()));
+///   await expectLater(tester, meetsGuideline(androidTapTargetGuideline));
+///   handle.dispose();
+/// });
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///  * [androidTapTargetGuideline]: which provides android specific guidance
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:81]: (here and everywhere): Our style is to use `,` instead of `:` before the `which`.

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:81]: android -> Android

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:81]: android -> Android

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:81]: maybe more specific: `which checks that tappable nodes have a minimum size of 48 by 48 pixels.`

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:81]: maybe more specific: `which checks that tappable semantic nodes have a minimum size of 48 by 48 pixels.`

File: packages/flutter_test/lib/src/accessibility.dart [hunk 2]
```diff
@@ -58,6 +58,34 @@ class Evaluation {
...
+///    for tappable widgets.
+///  * [iOSTapTargetGuideline]: which provides ios specific guidance for
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:83]: ios -> iOS

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:83]: ios -> iOS

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:83]: `which checks that tappable nodes a minimum size of 44 by 44 pixels.`

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:83]: `which checks that tappable semantic nodes a minimum size of 44 by 44 pixels.`

File: packages/flutter_test/lib/src/accessibility.dart [hunk 3]
```diff
@@ -58,6 +58,34 @@ class Evaluation {
...
+///    tappable widgets.
+///  * [textContrastGuideline]: which provides guidance for text contrast
+///    requirements specified by [WCAG](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html#contrast-ratiodef)
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:86]: missing `.` at the end of this one.

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:86]: missing `.` at the end of this one.

File: packages/flutter_test/lib/src/accessibility.dart [hunk 4]
```diff
@@ -73,6 +101,14 @@ abstract class AccessibilityGuideline {
...
 ///
 /// Each platform defines its own guidelines for minimum tap areas.
+///
+/// See also:
+///  * [AccessibilityGuideline]: which provides a general overview of
+///    accessibility guidelines and how to use them.
+///  * [androidTapTargetGuideline]: which provides android specific guidance
+///    for tappable widgets.
+///  * [iOSTapTargetGuideline]: which provides ios specific guidance for
+///    tappable widgets.
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:111]: maybe use the more specific suggestion from above here as well.

Comment (goderbauer) [inline packages/flutter_test/lib/src/accessibility.dart:111]: maybe use the more specific suggestion from above here as well.

File: packages/flutter_test/lib/src/matchers.dart [hunk 5]
```diff
@@ -663,12 +663,16 @@ Matcher matchesSemantics({
...
 ///   * [textContrastGuideline], for WCAG minimum text contrast guidelines.
+///   * [labeledTapTargetGuideline], for enforcing labels on tappable areas.
 AsyncMatcher meetsGuideline(AccessibilityGuideline guideline) {
   return _MatchesAccessibilityGuideline(guideline);
 }
 
 /// The inverse matcher of [meetsGuideline].
 ///
+/// This matcher is used for verifying accessibility guideline does flag the
+/// violations.
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/matchers.dart:674]: not sure what this sentence is supposed to say...

Comment (goderbauer) [inline packages/flutter_test/lib/src/matchers.dart:674]: not sure what this sentence is supposed to say...

**Review by goderbauer — COMMENTED** [review]
Thanks!

File: packages/flutter_test/lib/src/matchers.dart (hunk 5 above)
Comment (chunhtai) [reply packages/flutter_test/lib/src/matchers.dart:674]: it probably doesn't worth mention it. I will remove it

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**